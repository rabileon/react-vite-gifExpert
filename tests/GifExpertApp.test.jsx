import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  const setup = () => {
    render(<GifExpertApp />);
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form', { name: 'form' });
    return {
      input,
      form,
    };
  };

  test('sould match the snapshot', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('Agregar categorias', () => {
    const { input, form } = setup();

    fireEvent.input(input, { target: { value: 'Dragon Ball' } });
    fireEvent.submit(form);

    expect(screen.getAllByRole('heading', { level: 3 })).toHaveLength(1);
  });

  test('No agregar categorias repetidas', () => {
    const { input, form } = setup();

    fireEvent.change(input, { target: { value: 'One Punch' } });
    fireEvent.submit(form);

    fireEvent.change(input, { target: { value: 'One Punch' } });
    fireEvent.submit(form);

    expect(screen.getAllByRole('heading', { level: 3 })).toHaveLength(1);
  });
});
