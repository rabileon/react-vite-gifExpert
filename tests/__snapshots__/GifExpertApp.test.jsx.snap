// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> sould match the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <button>
    Agregar
  </button>
</div>
`;
